lib/multiplexer/wrappers.go

```diff
@@ -61,6 +69,28 @@ func (c *Conn) Protocol() int {
 	return c.protocol
 }
 
+// Detect detects the connection protocol by peeking into the first few bytes.
+func (c *Conn) Detect() (int, error) {
```

Comment:
while you're here, can you create a new type for these protocol numbers? something like `type Protocol int`